Barotrauma/BarotraumaClient/ClientSource/GUI/GUITextBox.cs

```diff
@@ -838,6 +927,53 @@ public void SelectAll()
             CalculateSelection();
         }
 
+        private static int FindStopSymbolLeft(string text, int idx)
+        {
+            // special case: if cursor at the start of the line, move to the end of previous line
+            if (idx > 0 && text[idx - 1] == '\n')
```

Comment:
Accessing the string array without first checking whether the index is within the valid range will throw an index out of range exception with "incorrect" or unexpected parameters, like `FindStopSymbolLeft("Test", 100)`.

Reply:
Also note that it might be ok to crash there, if it shouldn't be that case that the index can be outside of the string array, but I think it's better to make that explicit: throw a custom exception or a console error with some debug info about the text and the index. Currently it looks to me that it just happens to work, because the text and the index are in sync, because we use the Text and TextCaret fields. And if we just use them, I'd maybe just remove the parameters from the methods, make them non-static, and use the fields. In that case, there wouldn't be need for any extra error handling.